repository/config_git.go

```diff
@@ -0,0 +1,185 @@
+package repository
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/blang/semver"
+	"github.com/pkg/errors"
+)
+
+type gitConfig struct {
+	version *semver.Version
+	execFn  func(args ...string) (string, error)
+}
+
+func NewGitConfig(repo *GitRepo, global bool) *gitConfig {
+	version, _ := repo.GitVersion()
```

Comment:
Ignored this error in this draft, but we should not. Normally we should either we panic or return an error, but that will change the `LocalConfig()` `GlobalConfig()` signatures to `LocalConfig() (Config, error)`

Reply:
why not have the version checked at the same place as before ?

Reply:
Because `execFn` passes `config` arg by default and we can not use it to get the version :|, one other possible thing would be to get the version at the moment we create the `GitRepo` object and we passe it by parameter in `newGitConfig` function

Reply:
If i don't passe the `config` as arg i should have a boolean field storing the type of the struct (global/local) and well finish with a if/else situation in all the implementations

Reply:
Another way would to store the `repo.GitVersion` as a field in `GitConfig` struct

Reply:
what if you just store the flag as a string (`--local` or `--global`) ? You will then just have to insert it where needed.